drv/stm32xx-sys/src/main.rs

```diff
@@ -255,20 +395,241 @@ impl idl::InOrderSysImpl for ServerImpl<'_> {
+                                        w.bits(new_value)
+                                    }
+                                });
+                            }
+                        }
+                    }
+                }
+                Ok(())
+
+            } else {
+                // Suppress unused variable warnings (yay conditional
+                // compilation)
+                let _ = (rm, enable_mask, disable_mask);
+
+                // Fault any clients who try to use this in an image where it's
+                // not included.
+                Err(idol_runtime::ClientError::UnknownOperation.fail())
+            }
+        }
+    }
+
+    fn gpio_irq_configure(
+        &mut self,
+        rm: &RecvMessage,
+        mask: u32,
+        rising: bool,
+        falling: bool,
+    ) -> Result<(), RequestError<core::convert::Infallible>> {
+        // We want to only include code for this if exti is requested.
+        // Unfortunately the _operation_ is available unconditionally, but we'll
+        // fault any clients who call it if it's unsupported (below).
+        cfg_if! {
+            if #[cfg(feature = "exti")] {
+
+                for (i, entry) in generated::EXTI_DISPATCH_TABLE.iter().enumerate() {
+                    // Only use populated rows in the table
+                    if let Some((_, tid, entry_mask)) = entry {
+                        // Operate only on rows assigned to the sending task
+                        // _and_ that are relevant based on the arguments.
+                        if tid.index() == rm.sender.index() && mask & entry_mask != 0 {
+                            // Set or clear Rising Trigger Selection
+                            // Register bit according to the rising flag
+                            self.exti.rtsr1.modify(|r, w| {
+                                let new_value = if rising {
+                                    r.bits() | (1 << i)
+                                } else {
+                                    r.bits() & !(1 << i)
+                                };
+                                unsafe {
+                                    w.bits(new_value)
+                                }
+                            });
+                            // Set or clear Falling Trigger Selection
+                            // Register bit according to the rising flag
+                            self.exti.ftsr1.modify(|r, w| {
+                                let new_value = if falling {
+                                    r.bits() | (1 << i)
+                                } else {
+                                    r.bits() & !(1 << i)
+                                };
+                                unsafe {
+                                    w.bits(new_value)
+                                }
+                            });
+                        }
+                    }
+                }
+                Ok(())
```

Comment:
again, maybe it's worth reply-faulting if the mask doesn't actually include EXTI notifications?